src/Spanner/Session/CacheSessionPool.php

```diff
@@ -257,6 +268,11 @@ public function acquire($context = SessionPoolInterface::CONTEXT_READ)
             $session = $this->waitForNextAvailableSession();
         }
 
+        if ($this->deleteQueue) {
+            $this->deleteSessions($this->deleteQueue);
```

Comment:
Is it reasonable for this call to be inside `manageSessionsToDelete`? That would remove the need for the `deleteQueue` and (I think) mean that `deleteSessions` will be called earlier